omod/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProviderWebTest.java

```diff
@@ -550,4 +566,94 @@ private void verifyUri(String uri) throws Exception {
 		assertThat(results.getEntry().get(0).getResource(), notNullValue());
 		assertThat(results.getEntry().get(0).getResource().getIdElement().getIdPart(), equalTo(PATIENT_UUID));
 	}
+	
+	@Test
+	public void createPatient_shouldCreatePatient() throws Exception {
+		String jsonPatient;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_CREATE_PATIENT_PATH)) {
+			Objects.requireNonNull(is);
+			jsonPatient = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+
+		org.hl7.fhir.r4.model.Patient patient = new org.hl7.fhir.r4.model.Patient();
+		patient.setId(PATIENT_UUID);
+
+		when(patientService.create(any(org.hl7.fhir.r4.model.Patient.class))).thenReturn(patient);
+
+		MockHttpServletResponse response = post("/Patient").jsonContent(jsonPatient)
+		        .accept(FhirMediaTypes.JSON).go();
+
+		assertThat(response, isCreated());
+	}
+
+	@Test
+	public void updatePatient_shouldUpdateExistingPatient() throws Exception {
+		String jsonPatient;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_PATIENT_PATH)) {
+			Objects.requireNonNull(is);
+			jsonPatient = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+
+		org.hl7.fhir.r4.model.Patient patient = new org.hl7.fhir.r4.model.Patient();
+		patient.setId(PATIENT_UUID);
+
+		when(patientService.update(anyString(), any(org.hl7.fhir.r4.model.Patient.class))).thenReturn(patient);
+
+		MockHttpServletResponse response = put("/Patient/" + PATIENT_UUID).jsonContent(jsonPatient)
+				.accept(FhirMediaTypes.JSON).go();
+
+		assertThat(response, isOk());
+	}
+
+	@Test
+	public void updatePatient_shouldErrorForNoId() throws Exception {
+		String jsonPatient;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_PATIENT_NO_ID_PATH)) {
+			Objects.requireNonNull(is);
+			jsonPatient = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+
+		MockHttpServletResponse response = put("/Patient/" + PATIENT_UUID).jsonContent(jsonPatient)
+				.accept(FhirMediaTypes.JSON).go();
+
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentAsString(),
+				containsStringIgnoringCase("body must contain an ID element for update"));
+	}
+
+	@Test
+	public void updatePatient_shouldErrorForIdMissMatch() throws Exception {
```

Comment:
Typo in MissMatch, should be Mismatch instead

Reply:
@varung-31 First, thanks for taking the time to do these detailed reviews. Second, you might want to take advantage of GitHub's suggestion feature for some of these changes. E.g.,

```suggestion
	public void updatePatient_shouldRaiseExceptionOnIdMismatch() throws Exception {
```

Reply:
@ibacher  I was actually unaware about how to use it. I should maybe learn it to provide more helpful suggestions